src/actions/TemperatureSyncActions.js

```diff
@@ -85,9 +103,145 @@ const downloadLogs = sensor => async dispatch => {
+});
+const completeResettingLogFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_RESETTING_LOG_FREQUENCY,
+});
+const errorResettingLogFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.ERROR_RESETTING_LOG_FREQUENCY,
+});
+
+const resetLogFrequency = sensor => async dispatch => {
+  dispatch(startResettingLogFrequency());
+
+  const resettingLogFrequencyResult = await sensor.resetLogFrequency();
+
+  const { success } = resettingLogFrequencyResult;
+
+  if (success) dispatch(completeResettingLogFrequency());
+  else dispatch(errorResettingLogFrequency());
+
+  return new Promise(resolver => resolver(success));
+};
+
+const startSavingTemperatureLogs = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.START_SAVING_TEMPERATURE_LOGS,
+});
+const completeSavingTemperatureLogs = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_SAVING_TEMPERATURE_LOGS,
+});
+
+const createTemperatureLogs = sensor => dispatch => {
+  const { sensorLogs, location } = sensor;
+
+  // Only create TemperatureLogs for the greatest multiple of 6 SensorLogs,
+  // as each SensorLog is a 5 minute log, and each Temperature log a 30 minute log.
+  const iterateToo = sensorLogs.length - (sensorLogs.length % 6);
+  const sensorLogsToGroup = sensorLogs.sorted('timestamp').slice(0, iterateToo);
+  const groupedSensorLogs = chunk(sensorLogsToGroup, 6);
+
+  dispatch(startSavingTemperatureLogs(groupedSensorLogs.length));
+
+  UIDatabase.write(() => {
+    groupedSensorLogs.forEach(sensorLogGroup => {
+      const { isInBreach, mostRecentTemperatureBreach } = sensor;
+      const temperature = Math.min(...sensorLogGroup);
+      const timestamp = Math.min(...sensorLogGroup);
+
+      const newLog = createRecord(
+        UIDatabase,
+        'TemperatureLog',
+        temperature,
+        new Date(timestamp),
+        location
+      );
+
+      if (isInBreach) {
+        if (mostRecentTemperatureBreach?.willContinueBreach(newLog)) {
+          UIDatabase.update('TemperatureLog', { ...newLog, breach: mostRecentTemperatureBreach });
+        } else {
+          UIDatabase.update('TemperatureBreach', {
+            ...mostRecentTemperatureBreach,
+            endTimestamp: timestamp,
+          });
+        }
+      } else {
+        const breachConfigs = UIDatabase.objects('TemperatureBreachConfiguration');
+        breachConfigs.some(breachConfig => breachConfig.createBreach(UIDatabase, location, newLog));
+      }
+
+      UIDatabase.delete('SensorLog', groupedSensorLogs);
+    });
+  });
+
+  dispatch(completeSavingTemperatureLogs());
+
   return Promise.resolve();
 };
 
+const errorNoSensors = () => ({ type: TEMPERATURE_SYNC_ACTIONS.ERROR_NO_SENSORS });
+const startSync = () => ({ type: TEMPERATURE_SYNC_ACTIONS.START_SYNC });
+const completeSync = () => ({ type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_SYNC });
+
+const syncTemperatures = () => async (dispatch, getState) => {
+  const { isSyncing } = getState();
```

Comment:
this line should be `const { temperatureSync } = getState()` then `const {isSyncing} = temperatureSync` -- fixed in #2650